Migrate Tagit initialisation to a Stimulus Controller

lb- opened the pull request:
- Fixes #10100
- Does not migrate away from using a jQuery UI widget
- Preserves the existing global `initTagField` for now (can be removed in a future release)
- [x] Fix unit tests (python)
- [x] Add release considerations
- [x] Move `domReady` to shared util

lb- commented:
Ready for review, I plan to pull out the `domReady` promise util out to a shared util but ready for feedback from someone in the core team.

Would love your thoughts @Scotchester  or @thibaudcolas  - hopefully this is not a too controversial one.

As per original issue raised, I opted to not find a jQuery UI tag replacement yet, better we do that as an isolated bit of work. Thankfully this pulls out a large amount of CSP non-compliance and hopefully is a much simpler/secure (escpaing handled 100% at the widget side, no need to worry about it on the template side) way to provide the taggit widget.

nmorduch commented on docs/releases/5.0.md:
````diff
@@ -141,6 +141,40 @@ If there are custom styles in place for the `ModelAdmin`'s header content or mor
+#### Old syntax
+
+```html
+<input id="id_tags" type="text" value="popular, technology" hidden />
````
Nitpicky small comment: I would find this easier to compare if there were line breaks like in the example for the new syntax

lb- replied:
Good call. I originally had the others all in one line but after I ran prettier it wrapped the below attributes. 

I'll make the old syntax match this also.

lb- replied:
This is done. Thanks @nmorduch

lb- commented:
Rebased, fixed documentation concern, added an extra recommendation to use the `AdminTagWidget` instead. Additionally moved the `data-controller` declaration to the widgets' init call. This makes more sense as it's a bit more core to the widget behaviour, similar to an id.

lb- commented:
Argh. Ok I'll try to fix this tomorrow - the testing regex to pull out attributes is to fragile. Moving the data-controller attribute is causing it to fail.

lb- commented:
Good to go again for review now

lb- commented:
Rebased with upgrade considerations moved to 5.1 release notes.

lb- commented:
Rebased & upgrade considerations moved to 5.1 release notes.

lb- commented on client/src/utils/domReady.ts:
```diff
@@ -0,0 +1,16 @@
+/**
+ * Returns a promise that resolves once the DOM is ready for interaction.
+ */
+const domReady = () =>
```
No problems.

lb- replied:
It shouldn't change how the function is used though, but yeah I'll add Async/await to this function.

lb- commented:
Thanks for the feedback @laymonage  - ready for another review when you have some time.

laymonage reviewed (approved):
Looking great, thanks! Love seeing the inline scripts removal.
